slsim/Sources/source.py

```diff
@@ -125,6 +137,80 @@ def kwargs_variability_extracted(self):
                         }
+
+            # Check if AGN model is used
+            elif any(
+                element in kwargs_variability_list_agn
+                for element in list(self.kwargs_variability)
+            ):
+
+                z = self.source_dict["z"]
+                if self.cosmo is None:
+                    raise ValueError(
+                        "Cosmology cannot be None for AGN class. Please"
+                        "provide a suitable astropy cosmology."
+                    )
+
+                else:
+                    # Get the dict object from the astropy.table.column
+                    agn_kwarg_dict = self.source_dict["kwargs_agn_model"].data[0][0]
+
+                    # Populate "None" for optional keys related to drawing random AGN
+                    random_kwargs = {"random_seed": None, "input_agn_bounds_dict": None}
+                    for opt_key in random_kwargs.keys():
+                        if opt_key in agn_kwarg_dict.keys():
+                            random_kwargs[opt_key] = agn_kwarg_dict[opt_key]
+                            del agn_kwarg_dict[opt_key]
+
+                    # Create the agn object
+                    self.agn_class = agn.RandomAgn(
+                        self.source_dict["i_band_mag"],
+                        self.source_dict["z"],
+                        cosmo=self.cosmo,
+                        random_seed=random_kwargs["random_seed"],
+                        input_agn_bounds_dict=random_kwargs["input_agn_bounds_dict"],
+                        **agn_kwarg_dict
+                    )
+
+                    # Get mean mags across LSST filters
+                    # Fix to eventually give the user power to choose other surveys
+                    mean_magnitudes = self.agn_class.get_mean_mags("lsst")
+
+                    # These are lsst filters, hard coded in for now. Fix later.
+                    filters = ["u", "g", "r", "i", "z", "y"]
+                    for index, band in enumerate(filters):
+
+                        # Define name for point source mags
+                        filter_name = "ps_mag_" + band
+
+                        # Again, lsst filters hard coded in. Fix later.
+                        self.agn_class.variable_disk.reprocessing_kwargs[
+                            "speclite_filter"
+                        ] = ("lsst2023-" + band)
+
+                        self.agn_class.variable_disk.driving_signal_kwargs[
+                            "mean_magnitude"
+                        ] = mean_magnitudes[index]
+
+                        reprocessed_lightcurve = reprocess_with_lamppost_model(
+                            self.agn_class.variable_disk
+                        )
+
+                        times = reprocessed_lightcurve["MJD"]
+                        magnitudes = reprocessed_lightcurve["ps_mag_lsst2023-" + band]
+
+                        new_column = Column([float(min(magnitudes))], name=filter_name)
```

Comment:
In case of supernovae, supernovae sample does not have ps_mag_i or simply peak magnitude so we store them from the lightcurve. But, in quasar case, quasar sample have ps_mag_i and you use this magnitude from the source _dict while getting the quasar lightcurve. So, I think it is not necessary to store peak magnitude from the lightcurve and lines 202 to 206 can be excluded.

Reply:
Got it! I just copied these lines of code thinking you had to create a new Column for each band

Reply:
I did have to include the creation of the new columns so that the method point_source_magnitude could find the column named "ps_mag_"+band.